File: typescript/src/errors/VerificationError.ts
import {
  ErrorCode,
  ParmanaError,
} from "./ParmanaError.js";

/**
 * Raised when verification fails.
 */
export class VerificationError extends ParmanaError {
  constructor(
    message: string,
    options?: {
      requestId?: string;
      cause?: unknown;
    },
  ) {
    super({
      code: ErrorCode.VERIFICATION_ERROR,

      message,

      ...(options?.requestId !== undefined && {
        requestId: options.requestId,
      }),

      ...(options?.cause !== undefined && {
        cause: options.cause,
      }),
    });

    this.name = "VerificationError";
  }
}

File: python/parmana/errors/verification_error.py
from __future__ import annotations

from .api_error import ParmanaError


class VerificationError(ParmanaError):
    """
    Raised when verification fails. Mirrors
    typescript/src/errors/VerificationError.ts.
    """

    def __init__(
        self,
        message: str,
        *,
        request_id: str | None = None,
    ) -> None:
        super().__init__(
            message,
            code="VERIFICATION_ERROR",
            request_id=request_id,
        )
